cmd/opm/alpha/template/composite.go

```diff
@@ -31,11 +34,23 @@ and a 'composite template' file`,
 		Args: cobra.MaximumNArgs(0),
 		Run: func(cmd *cobra.Command, args []string) {
 			containerTool = "docker"
-			catalogData, err := os.Open(catalogFile)
+			var tempCatalog io.ReadCloser
```

Comment:
Probably a stupid question w.r.t. golang and deferred calls, but is there any reason that we couldn't do 
```golang
var catalogData io.ReadCloser
```
here, and in the conditionals, assign to it/defer closure to it as needed? 

If it's imposed by the language, fine, but L53 feels like we could optimize it away with this approach.

Reply:
/resolve